pkg/bundle/internal/target/target.go

```diff
@@ -254,6 +258,34 @@ func (r *Reconciler) needsUpdate(ctx context.Context, kind Kind, obj *metav1.Par
 		needsUpdate = true
 	}
 
+	// Find keys that are not present in the target but should be
+	for annotationKey, annotationValue := range annotations {
+		if obj.GetAnnotations()[annotationKey] != annotationValue {
+			needsUpdate = true
+		}
+	}
+	for labelKey, labelValue := range labels {
+		if obj.GetLabels()[labelKey] != labelValue {
+			needsUpdate = true
+		}
+	}
+
+	// Find keys present in the target that should be removed
+	for annotationKey := range obj.GetAnnotations() {
+		if annotationKey == trustapi.BundleHashAnnotationKey {
+			continue
+		} else if _, exists := annotations[annotationKey]; !exists {
+			needsUpdate = true
+		}
+	}
+	for labelKey := range obj.GetLabels() {
+		if labelKey == trustapi.BundleLabelKey {
+			continue
+		} else if _, exists := labels[labelKey]; !exists {
+			needsUpdate = true
+		}
+	}
+
```

Comment:
```suggestion
```

Since we are using server-side apply (SSA) in trust-manager, checking for labels/annotations to remove should not be required. But since we have this "needsUpdate" logic, and don't patch targets unconditionally, we need to create "smarter" logic. Probably based on the managed fields.

Other operators may add labels/annotations to targets, so we cannot just assume trust-manager is managing ALL labels/annotations.

Maybe @inteon has a good solution to this?

Reply:
Yep, updated the hash function to take target annotations/labels key & values into consideration if they exist, then the hash should always be different for any combination but also ignore any labels/annotations added by third parties.